Fix typo in zero tensor creation and add background correction test

E-Rum opened the pull request.

PicoCentauri commented on tests/test_potentials.py:
```diff
@@ -640,3 +640,15 @@ def test_inverserp_vs_spline(exponent, smearing):
+    if exponent >= 3:
+        assert torch.allclose(bg, torch.tensor([0.0], dtype=dtype))
+    else:
+        assert not torch.allclose(bg, torch.tensor([0.0], dtype=dtype))
```
I would paste the equation here!

PicoCentauri reviewed (changes requested):
Very good catch! Thanks for the fix and also to @tulga-rdn for probably fining it 

I think we should try to add similar tests for torch script as for the calculators

https://github.com/lab-cosmo/torch-pme/blob/30aec6b6384a995619e32cc10eda6b97fe1a11d8/tests/calculators/test_workflow.py#L113-L139

to

https://github.com/lab-cosmo/torch-pme/blob/30aec6b6384a995619e32cc10eda6b97fe1a11d8/tests/test_potentials.py#L396-L442

and add a new element for `p>3`! The tests are basically scripting writing to a file and rereading!

E-Rum commented:
I addressed most of the comments in the new commit. I think we might get away without adding the same tests from calculators to potentials by simply broadening the returns from the JIT function check in potentials: now we also return the self-contribution and background correction. I checked that without these, a call error with the non-existing 'torch_zero' would be raised. Additionally, with this new test, I also caught a new annotation problem in `SplinePotential`, which I have also resolved
